### PR TITLE
updating useAsyncStorage syntax

### PR DESCRIPTION
**Pull request by idkjs**
Lets user do:

#### `ReactNativeAsyncStorage.asyncStorageState`

```reason
type asyncStorageState = {
  getItem: unit => Js.Promise.t(Js.Null.t(string)),
  setItem: string => Js.Promise.t(unit),
  mergeItem: string => Js.Promise.t(unit),
  removeItem: unit => Js.Promise.t(unit),
};

let key = "@@Key"
 let ReactNativeAsyncStorage.{getItem, setItem} =
    ReactNativeAsyncStorage.useAsyncStorage("@@Key");
```

Instead of 

```reason
 let useAsyncStorage=
    ReactNativeAsyncStorage.useAsyncStorage("@@Key");

let getItem =useAsyncStorage##getItem;
```

See updated [README.md](https://github.com/reason-react-native/async-storage/blob/1678c86/README.md#L49)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,6 +134,104 @@ unit => unit
...
+let key = "@@KEY";
+
+
+[@react.component]
+let app = () => {
+  let (value, setValue) = React.useState(() => "");
+  let ReactNativeAsyncStorage.{getItem, setItem} =
+    ReactNativeAsyncStorage.useAsyncStorage("@@KEY");
+
+  let getItem = () => useAsyncStorage##getItem();
```
Comment (tatchi) [inline README.md]: Shouldn't this line be removed?

**Review by tatchi — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (idkjs) [reply README.md]: Yes. Done. Thank you.

**Review by tatchi — APPROVED** [review]

**Review by MoOx — CHANGES_REQUESTED** [review]
Thanks for this PR! Could the example be more straight forward without styles for example? To avoid unecessary noise so the developer can focus on the actual interesting code, thanks!

**Comment by idkjs** [comment]
@MoOx https://github.com/reason-react-native/async-storage/pull/4#pullrequestreview-564167837 done

**Comment by MoOx** [comment]
I removed example as they were way to complex to quickly illustrate how to use the hook. If you want to another PR, but without UI & faker & unecessary stuff for the developer to understand the core of the usage.

**Comment by MoOx** [comment]
I rolled back the change and added record syntax as `useAsyncStorage_` with a deprecation notice for old syntax.

**Pull request merged**